kitchensink/src/app/common/PageElements/FoundationElements/ColorRow/ColorRow.jsx

```diff
@@ -0,0 +1,167 @@
+                  bordered={props.rowHeader.colors[0].bordered}
+                  stackPosition={props.rowHeader.colors[0].stack}
+                  infoBox={{ infoTitle: props.rowHeader.colors[0].colorName, infoLabel: props.rowHeader.colors[0].infoLabel }}
+                  key={props.rowHeader.colors[0].name}
+                  centered={false}
+                />
+              </td>
+            )
+          }
+          {
+            props.preColumns > 0
+            && <td colSpan={props.preColumns} />
+          }
+          {
+            props.rowData.map((element) => (
+              <td key={`${element.colors[0].name}`}>
+                {
+                  element.colors.map((color) => (
+                    <ColorBox
+                      variableName={color.name}
+                      hexValue={color.hex}
+                      size={props.rowSize}
+                      darkFont={color.darkFont}
+                      bordered={color.bordered}
+                      stackPosition={color.stack}
+                      infoBox={(function () {
+                        if (props.showInfo) {
+                          if (props.showInfo === 'fullInfo') {
+                            return { infoTitle: color.colorName, infoText: color.name, infoLabel: color.infoLabel };
+                          }
+                          if (props.showInfo === 'onlyTitles') {
+                            if (color.colorName) {
+                              return { infoTitle: color.colorName };
+                            }
+                          }
+                        }
+                        return {};
+                      }())}
+                      key={color.name}
+                    />
+                  ))
+                }
+              </td>
+            ))
+          }
+          {
+            // This is used to create column padding and is set in props.columns
+            (props.columns - (props.rowHeader ? props.rowData.length + 1 : props.rowData) - (props.preColumns ? props.preColumns : 0)) > 0
+            && <td colSpan={(props.columns - (props.rowHeader ? props.rowData.length + 1 : props.rowData) - (props.preColumns ? props.preColumns : 0))} />
+          }
+        </tr>
+      </tbody>
+    </table>
+    {
+      (props.description)
+      && (
+        <div styleName="descriptionCell">
+          {props.description}
+        </div>
+      )
+    }
+  </div>
+);
+
+ColorRow.displayName = 'ColorRow';
+
+ColorRow.propTypes = {
+  columns: PropTypes.number,
+  description: PropTypes.node,
+  preColumns: PropTypes.number,
+  rowData: PropTypes.arrayOf(
+    PropTypes.shape({
+      colors: PropTypes.arrayOf(
+        PropTypes.shape({
+          bordered: PropTypes.bool,
+          colorName: PropTypes.string,
+          darkFont: PropTypes.bool,
+          hex: PropTypes.string,
+          infoLabel: PropTypes.string,
+          name: PropTypes.string,
+          stack: PropTypes.string,
```

Comment:
Could you store this object into a separate object please.

So you can reuse it further down.